### PR TITLE
Handle field name overlaps

### LINKED ISSUE #11251: [codegen/go]: Property named "elementType" conflicts with built-in ElementType method
### What happened?

https://github.com/pulumi/pulumi-google-native/pull/733 added a new resource with a nested property called `elementType`. This generated the following code (irrelevant code snipped):
```go
type EnterpriseCrmEventbusProtoNodeIdentifierArgs struct {
	// Destination node where the edge ends. It can only be a task config.
	ElementType EnterpriseCrmEventbusProtoNodeIdentifierElementTypePtrInput `pulumi:"elementType"`
}

func (EnterpriseCrmEventbusProtoNodeIdentifierArgs) ElementType() reflect.Type {
	return reflect.TypeOf((*EnterpriseCrmEventbusProtoNodeIdentifier)(nil)).Elem()
}
```

This results in the following compilation error:
```
go/google/integrations/v1alpha/pulumiTypes.go:3509:53: field and method with the same name ElementType
	go/google/integrations/v1alpha/pulumiTypes.go:3506:2: other declaration of ElementType
```

### Steps to reproduce

Add a resource property called `elementType`. The generated Go SDK will have a compilation error.

### Expected Behavior

Ideally, the code generator would disambiguate the property name (e.g., `ElementType_`). Barring that, the code generator should at least issue a warning that `ElementType` is a reserved word in the Pulumi Go SDK.

### Actual Behavior

Invalid code was generated

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by iwahbe**
Fixes #11251

### CODE REVIEW

**Review by lblackstone — APPROVED** [review]
LGTM -- thanks for the quick fix!

File: pkg/codegen/go/utilities.go [hunk 1]
```diff
@@ -120,3 +137,47 @@ func enumTitle(s string) string {
...
+				postfix = "Str"
+			case schema.BoolType:
+				postfix = "Bool"
+			case schema.IntType:
+				postfix = "Int"
+			case schema.NumberType:
+				postfix = "Num"
+			}
+		}
+	}
```
Comment (AaronFriel) [inline pkg/codegen/go/utilities.go:172]: Could we simplify this to adding a `_` suffix? Each object type can only have a single field named `ElementType` for example, I don't see what we gain by adding type info in the field name.

Comment (iwahbe) [reply pkg/codegen/go/utilities.go:172]: We could, I just thought that this looked more natural. Its inline with how we deal with `*schema.Object` and `*schema.Resource` name conflicts.

File: pkg/codegen/go/utilities.go [hunk 2]
```diff
@@ -40,6 +42,21 @@ func isReservedWord(s string) bool {
 	}
 }
 
+// isReservedResourceField returns true if s would conflict with a method on a generated
+// resource.
+func isReservedResourceField(resourceName, s string) bool {
+	switch s {
+	case "ID", "URN", "GetProvider", "ElementType":
```
Comment (Frassle) [inline pkg/codegen/go/utilities.go:49]: Are there any Go reserved words we should be checking here as well?

**Comment by kpitzen** [comment]
bors retry

**Comment by RobbieMcKinstry** [comment]
bors retry

**Comment by kpitzen** [comment]
bors retry

**Review by lblackstone — APPROVED** [review]

**Comment by kpitzen** [comment]
bors merge

**Comment by kpitzen** [comment]
bors retry

**Comment by kpitzen** [comment]
bors retry

**Comment by kpitzen** [comment]
bors retry

**Pull request merged**